### ISSUE TITLE
More Style Options for Text Editor/Doc Incorrect?

### ISSUE THREAD

**Issue by DapperMongoose** [cogentcore/core#1278]
### Describe the feature

Hello again!  I've started digging around and working with this lovely framework and I'm really liking it so far.  It has a ton of potential!  

I'm currently trying to display output that would theoretically be streamed in over time as messages arrive to the client I'm writing.  Like a chat program.  From what I've read in the docs and tried it seems like the text editor set to read only is exactly the widget for this text (I imagine that I'd stream updates to the buffer as appropriate)

However, for this case I don't really want the line numbers.  I'd really like either a less opinionated (style wise) widget for something like this, but I'd also be happy if I could turn off the numbers as well.  

I thought that perhaps I could hide them by setting the color when I saw:
```
// LineNumberColor is the color used for the side bar containing the line numbers.
// This should be set in Stylers like all other style properties.
LineNumberColor image.Image
```
	
However I can't seem to figure out how I am supposed to do that, is the doc string for this no longer correct?  Am I maybe barking up the wrong tree here?

**Comment by kkoreilly** [comment]
Thank you for your feedback, and apologies for the unclear documentation. The best way to turn off line numbers for a text editor is this:

```go
ed.Buffer.Options.LineNumbers = false
```

We can definitely improve the documentation soon to make that clearer. The docs for `LineNumberColor` are not wrong, but that property is rarely relevant (it is already transparent by default). The default text editor styles are somewhat opinionated, but they are fully customizable, so once we elaborate the documentation, that should be fine.

For your specific use case, you may also be interested in `texteditor.OutputBuffer` (see [here in our terminal emulator](https://github.com/cogentcore/cogent/blob/main/terminal/app.go#L187) for an example; I can also provide more help if you need it).

If it is plaintext/HTML for a chat program, you could also use a `core.Text` widget or `htmlcore` and call `Update()` to keep it in sync (see [here in our AI client](https://github.com/cogentcore/cogent/blob/main/ai/ai.go#L197) for an HTML example). I am happy to give guidance for any of these approaches.

**Comment by ghost** [comment]
Ahh, that did the trick!  I was looking at the editor itself and didn't think to check the buffer itself.  

Honestly the docs so far have been the biggest struggle.  As I slowly wrap my head around the Cogent way of doing things I'm continually impressed at how clean and sane it is.  

I'm a back end dev and I'm usually really frustrated by how awkward and convoluted front end frameworks are, I'm having a pretty good time with Cogent so far though!

**Comment by kkoreilly** [comment]
That's great to hear! Improving the documentation is definitely very high on our priority list.

**Comment by ghost** [comment]
OK, I was able to progress further, but now I have two more things that are stumping me.  For a quick test I just wired up the input (text box) to the same io.Pipe() as the OutputBuffer is reading,  that example was great.  

The output editor is currently set non selectible, non hoverable, and read only.  The idea being that it will display incoming messages.

I set StartFocus() on my input box and on launch it has focus as expected.  I can immediately type and hit enter (I tried both with OnKeyChord switching on the keymap.Enter value,  and with OnChange) and I can see the output in the output box.  After I write to the pipe in the handler I call SetText to clear the input text and SetFocusEvent() to return focus there.

Problem is the focus doesn't switch back, I'm actually not completely sure where it goes.  It may briefly be on that output editor because I can see the text cursor (not animated) on the new line.  I can manually set focus back to the input box and it behaves normally.  I can also select the output editor and see the text cursor animate.

Setting abilities.Focusable to false doesn't seem to help either.

This is the focus debug trace for the following steps:
1. Start app
2. Type foo into text input box and hit enter
3. Manually change focus back to input box

```/Cogent Core scene SetFocusEvent: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocusEvent: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene Already Focus: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocusEvent: Failed /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocusEvent: <nil>
/Cogent Core scene Losing focus: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocusEvent: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocusEvent: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene Already Focus: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocusEvent: Failed /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocus: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene Already Focus: /Cogent Core scene…rame-1/text-field-0
/Cogent Core scene SetFocus: Failed /Cogent Core scene…rame-1/text-field-0
```

The layout of the app so far is the core body with two frames inside of it.  One frame holds the output text editor, the second holds the input text box.

**Comment by ghost** [comment]
I think I figured it out.  I was running afoul of something else handling the enter keypress after the KeyChord handler.  Setting e.SetHandled() as the last step of the event handler has put a stop to that.

**Comment by kkoreilly** [comment]
Yes, the text field event handler is handling the KeyChord event later and defocusing the text field at that point. Because that handler also calls `e.SetHandled()`, it doesn't work to do `OnFinal`, so calling `e.SetHandled()` in your event handler is the best approach as you said.
